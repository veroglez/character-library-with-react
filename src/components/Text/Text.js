import { bool, node, string } from 'prop-types';
import React from 'react';
import scss from './Text.module.scss';

const determineStyle = ({
  caption, headline, subtitle,
}) => {
  if (headline) return scss.headline;
  if (subtitle) return scss.subtitle;
  if (caption) return scss.caption;
  return scss.body;
};

const Text = ({
  caption, children, headline, subtitle, color, ...inherit
}) => (
  <p className={`${scss.text} ${determineStyle({ caption, headline, subtitle })}${inherit.styles ? ` ${inherit.styles}` : ''}${color ? ` ${scss[color]}` : ''}`}>
    {children}
  </p>
);

Text.propTypes = {
  caption: bool,
  color: string,
  children: node,
  headline: bool,
  subtitle: bool,
};

Text.defaultProps = {
  caption: false,
  color: undefined,
  children: undefined,
  headline: false,
  subtitle: false,
};

export default Text;
