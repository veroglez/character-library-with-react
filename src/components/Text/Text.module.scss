.text {
  margin: 0;
  font-family: "Roboto", sans-serif;
  color: black;
};

.body {
  font-size: 16px;
}

.headline {
  font-size: 24px;
  font-weight: 700;
  letter-spacing: -0.6px;
}

.subtitle {
  font-size: 15px;
  font-weight: 700;
  padding-bottom: 12px;
}

.caption {
  border: 2px solid;
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  padding: 3px 10px;
}

.yellow {
  color: #fde145;
  border-color: #fde145;
}

.green {
  color: #30c52e;
  border-color: #30c52e;
}

.orange {
  color: #fe7250;
  border-color: #fe7250;
}