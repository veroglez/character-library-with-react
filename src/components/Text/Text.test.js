import React from 'react';
import renderer from 'react-test-renderer';

import Text from './Text';

const TEXT = 'Lorem ipsum';
const COLOR = 'green';

describe('<Text>', () => {
  it('renders', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {headline}', () => {
    const tree = renderer.create(<Text headline />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {subtitle}', () => {
    const tree = renderer.create(<Text subtitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<Text color={COLOR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {children}', () => {
    const tree = renderer.create(<Text>{TEXT}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {inherit}', () => {
    const tree = renderer.create(<Text styles="style">{TEXT}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
