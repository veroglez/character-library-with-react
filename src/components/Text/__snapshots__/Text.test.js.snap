// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> renders 1`] = `
<p
  className="text body"
/>
`;

exports[`<Text> when {children} 1`] = `
<p
  className="text body"
>
  Lorem ipsum
</p>
`;

exports[`<Text> when {color} 1`] = `
<p
  className="text body green"
/>
`;

exports[`<Text> when {headline} 1`] = `
<p
  className="text headline"
/>
`;

exports[`<Text> when {inherit} 1`] = `
<p
  className="text body style"
>
  Lorem ipsum
</p>
`;

exports[`<Text> when {subtitle} 1`] = `
<p
  className="text subtitle"
/>
`;
